src/Routing/Url.php

```diff
@@ -0,0 +1,272 @@
+     * @return $this
+     */
+    public function setParams(array $params)
+    {
+        foreach ($params as $key => $value) {
+            $this->setParam($key, $value);
+        }
+
+        return $this;
+    }
+
+    /**
+     * Sets a query param
+     *
+     * @param string $key Query param name
+     * @param string|int|float $value Value
+     * @return $this
+     */
+    public function setQuery($key, $value)
+    {
+        if (!isset($this->url['?'])) {
+            $this->url['?'] = [];
+        }
+        $this->url['?'][$key] = $value;
+
+        return $this;
+    }
+
+    /**
+     * Sets multiple query params
+     *
+     * @param array $params Query params as key value list
+     * @return $this
+     */
+    public function setQueryParams(array $params)
+    {
+        if (!isset($this->url['?'])) {
+            $this->url['?'] = [];
+        }
+
+        $this->url['?'] = array_merge($this->url['?'], $params);
+
+        return $this;
+    }
+
+    /**
+     * Absolute URL
+     *
+     * @param bool $absolute Generate an absolute URL or not
+     * @return $this
+     */
+    public function setAbsolute($absolute = true)
+    {
+        $this->absolute = (bool)$absolute;
+
+        return $this;
+    }
+
+    /**
+     * To string
+     *
+     * @return string;
+     */
+    public function toString()
+    {
+        return Router::url($this->url, $this->absolute);
+    }
+
+    /**
+     * To array
+     *
+     * @return array
+     */
+    public function toArray()
+    {
+        return $this->url;
+    }
+
+    /**
+     * To string
+     *
+     * @return string String URL
+     */
+    public function __toString()
+    {
+        return $this->toString();
+    }
+
+    /**
+     * Whether a offset exists
+     *
+     * @link http://php.net/manual/en/arrayaccess.offsetexists.php
+     * @param mixed $offset An offset to check for.
+     * @return bool true on success or false on failure.
+     * The return value will be casted to boolean if non-boolean was returned.
+     */
+    public function offsetExists($offset)
+    {
+        return isset($this->url[$offset]);
+    }
+
+    /**
+     * Offset to retrieve
+     *
+     * @link http://php.net/manual/en/arrayaccess.offsetget.php
+     * @param mixed $offset The offset to retrieve.
+     * @return mixed Can return all value types.
+     */
+    public function offsetGet($offset)
+    {
+        if (isset($this->url[$offset])) {
+            return $this->url[$offset];
+        }
+
+        return null;
+    }
+
+    /**
+     * Offset to set
+     *
+     * @link http://php.net/manual/en/arrayaccess.offsetset.php
+     * @param mixed $offset The offset to assign the value to.
+     * @param mixed $value The value to set.
+     * @return void
+     */
+    public function offsetSet($offset, $value)
+    {
+        $this->url[$offset] = $value;
+    }
+
+    /**
+     * Offset to unset
+     *
+     * @link http://php.net/manual/en/arrayaccess.offsetunset.php
```

Comment:
Do we have a convention for that? I usually put things like that before params and returns.